### PR TITLE
Erica/radius

### PR DESCRIPTION
**Pull request by ericayin**

### CODE REVIEW

File: sage-ios/SAGE/SAGE/AddSchoolController.swift [hunk 1]
```diff
@@ -13,6 +13,8 @@ class AddSchoolController: FormController {
     var director: User?
     var location: CLLocation?
     var address: String?
+    var radius: CLLocationDistance? = nil
+    var radiusCenter: CLLocationCoordinate2D? = CLLocationCoordinate2D()
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolController.swift]: isnt radiusCenter the same as location.coordinate?

Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolController.swift]: also this doesnt need to be optional if your gonna initalize it

File: sage-ios/SAGE/SAGE/AddSchoolRadiusView.swift [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        
+        let circleCenter = self.radiusCenter
+        self.circle.fillColor = UIColor.colorWithIntRed(74, green: 144, blue: 226, alpha: 0.15)
+        self.circle.strokeColor = UIColor.lightBlueColor
+        self.circle.strokeWidth = 1
+        self.circle.position = circleCenter
+        self.circle.radius = CLLocationDistance(self.slider.value)
+        self.circle.map = self.mapView
+        
+        self.layoutSubviews()
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolRadiusView.swift]: dont need this

File: sage-ios/SAGE/SAGE/School.swift [hunk 3]
```diff
@@ -16,8 +16,9 @@ class School: NSObject, NSCoding {
     var director: User?
     var address: String?
     var studentCount: Int = 0
+    var radius: CLLocationDistance = -1
     
-    init(id: Int = -1, name: String? = nil, location: CLLocation? = nil, students: [User]? = nil, director: User? = nil, address: String? = nil, studentCount: Int = 0) {
+    init(id: Int = -1, name: String? = nil, location: CLLocation? = nil, students: [User]? = nil, director: User? = nil, address: String? = nil, studentCount: Int = 0, radius: CLLocationDistance = -1) {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/School.swift]: default the radius to 200

File: sage-ios/SAGE/SAGE/AddSchoolRadiusView.swift [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    
+    func radiusToString(radius: Float) -> String {
+        var index = 1
+        if (radius >= 10 && radius < 100) {
+            index = 2
+        } else if (radius >= 100) {
+            index = 3
+        }
+        let roundedValueString: String = (String(radius) as NSString).substringToIndex(index)
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolRadiusView.swift]: cant you just truncate to int soyou dont have to deal with any substring index stuff?

File: sage-ios/SAGE/SAGE/AddSchoolController.swift [hunk 5]
```diff
@@ -41,15 +44,34 @@ class AddSchoolController: FormController {
         self.navigationController?.pushViewController(tableViewController, animated: true)
     }
     
+    func radiusButtonTapped() {
+        if self.location == nil {
+            self.showAlertControllerError("Please choose a location first.")
+        } else {
+            if self.location?.coordinate != nil {
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolController.swift]: you know that self.location wont be nil at this point, so you dont need to make this nil check again

File: sage-ios/SAGE/SAGE/AddSchoolController.swift [hunk 6]
```diff
@@ -41,15 +44,34 @@ class AddSchoolController: FormController {
...
+                self.radiusCenter = self.location!.coordinate
+            }
+            let viewController = AddSchoolRadiusViewController(center: self.radiusCenter!, radius: self.radius)
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolController.swift]: is it possible to just pass in self.location.coordinate into the viewController? it doesnt seem like you need self.radiusCenter for anything else other than passing it as a param into the VC

File: sage-ios/SAGE/SAGE/AddSchoolRadiusViewController.swift [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+//
+//  Created by Erica Yin on 2/21/16.
+//  Copyright © 2016 Cal Blueprint. All rights reserved.
+//
+
+import Foundation
+
+class AddSchoolRadiusViewController: UIViewController {
+    
+    var schoolRadiusView: AddSchoolRadiusView?
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolRadiusViewController.swift]: doesnt need to be optional just initialize it here and in loadView just say self.view = self.schoolRadiusView

File: sage-ios/SAGE/SAGE/AddSchoolRadiusViewController.swift [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    }
+    
+    deinit {
+        NSNotificationCenter.defaultCenter().removeObserver(self)
+    }
+
+    override func loadView() {
+        super.loadView()
+        self.view = self.schoolRadiusView
+        let saveButton: UIBarButtonItem = UIBarButtonItem(title: "Finish", style: UIBarButtonItemStyle.Plain, target: self, action: "saveRadius:")
```
Comment (amillman) [inline sage-ios/SAGE/SAGE/AddSchoolRadiusViewController.swift]: put this saveButton stuff in viewDidLoad

**Comment by amillman** [comment]
make sure to pass in the school radius in createSchool params dictionary

**Comment by charlesx2013** [comment]
Please make better commit comments please QQ

**Pull request merged**